### PR TITLE
client request metrics should be registered only once

### PR DESCRIPTION
**Pull request by hongchaodeng**
This is one of the contention found in density tests.
@wojtek-t @gmarek @fgrzadkowski

### CODE REVIEW

**Comment by smarterclayton** [comment]
Yikes.

**Comment by smarterclayton** [comment]
ok to test

**Comment by wojtek-t** [comment]
LGTM - thanks!

File: pkg/client/unversioned/request.go [hunk 1]
```diff
@@ -48,6 +48,10 @@ import (
 // are therefore not allowed to set manually.
 var specialParams = sets.NewString("timeout")
 
+func init() {
+	metrics.Register()
```
Comment (lavalamp) [inline pkg/client/unversioned/request.go]: This is an improvement, but metrics itself should protect against double-registration IMO.

Comment (xiang90) [inline pkg/client/unversioned/request.go]: @lavalamp It does protect you from this. And in the k8s metrics pkg, it has a double protection by using sync.Once. But this change makes things easier to reason about.

Comment (lavalamp) [inline pkg/client/unversioned/request.go]: Ah, great!

Comment (wojtek-t) [inline pkg/client/unversioned/request.go]: Yeah - the code before was correct - it wasn't correctness improvement.

It is just that we avoid some unnecessary checks (under lock) when creating every single request.

**Pull request merged**